lib/pwwka/persistent_transmitter.rb

```diff
@@ -0,0 +1,73 @@
+require_relative "publish_options"
+
+begin  # optional dependency
+  require 'resque'
+  require 'resque-retry'
+rescue LoadError
+end
+
+module Pwwka
+  # Primary interface for sending messages.
+  #
+  # Example:
+  #
+  #     # Send a message, blowing up if there's any problem
+  #     Pwwka::Transmitter.send_message!({ user_id: @user.id }, "users.user.activated")
+  #
+  #     # Send a message, logging if there's any problem
+  #     Pwwka::Transmitter.send_message_safely({ user_id: @user.id }, "users.user.activated")
+  class PersistentTransmitter
+
+    extend Pwwka::Logging
+    include Pwwka::Logging
+
+    DEFAULT_DELAY_BY_MS = 5000
+
+    attr_reader :channel_connector
+    private_class_method :new
```

Comment:
ah, is this all it takes to make it private?

Reply:
yes but this won't work unfortunately. I can't call new from within the class method when I use this